Userland/Libraries/LibGfx/ImageFormats/WebPLoader.cpp

```diff
@@ -297,6 +298,352 @@ static ErrorOr<VP8LHeader> decode_webp_chunk_VP8L_header(WebPLoadingContext& con
+    while (code_lengths.size() < alphabet_size) {
+        auto symbol = TRY(code_length_code.read_symbol(bit_stream));
+
+        if (symbol < 16) {
+            // "Code [0..15] indicates literal code lengths."
+            dbgln_if(WEBP_DEBUG, "  append {}", symbol);
+            code_lengths.append(static_cast<u8>(symbol));
+            if (symbol != 0)
+                last_non_zero = symbol;
+            continue;
+        } else if (symbol == 17) {
+            // "Code 17 emits a streak of zeros [3..10], i.e., 3 + ReadBits(3) times."
+            auto nrepeat = 3 + TRY(bit_stream.read_bits(3));
+            dbgln_if(WEBP_DEBUG, "  repeat {} zeroes", nrepeat);
+            for (size_t j = 0; j < nrepeat; ++j)
+                code_lengths.append(0);
+            continue;
+        } else if (symbol == 18) {
+            // "Code 18 emits a streak of zeros of length [11..138], i.e., 11 + ReadBits(7) times."
+            auto nrepeat = 11 + TRY(bit_stream.read_bits(7));
+            dbgln_if(WEBP_DEBUG, "  Repeat {} zeroes", nrepeat);
+            for (size_t j = 0; j < nrepeat; ++j)
+                code_lengths.append(0);
+            continue;
+        } else {
+            // "Code 16 repeats the previous non-zero value [3..6] times, i.e., 3 + ReadBits(2) times."
+            VERIFY(symbol == 16);
+
+            auto nrepeat = 3 + TRY(bit_stream.read_bits(2));
+            dbgln_if(WEBP_DEBUG, "  repeat {} {}s", nrepeat, last_non_zero);
+
+            // This is different from deflate.
+            for (size_t j = 0; j < nrepeat; ++j)
+                code_lengths.append(last_non_zero);
+        }
+    }
+
+    if (code_lengths.size() != alphabet_size)
+        return Error::from_string_literal("Number of code lengths does not match the sum of codes");
+
+    return Compress::CanonicalCode::from_bytes(code_lengths);
+}
+
+// https://developers.google.com/speed/webp/docs/webp_lossless_bitstream_specification#622_decoding_of_meta_prefix_codes
+// The description of prefix code groups is in "Decoding of Meta Prefix Codes", even though prefix code groups are used
+// in regular images without meta prefix code as well ¯\_(ツ)_/¯.
+static ErrorOr<PrefixCodeGroup> decode_webp_chunk_VP8L_prefix_code_group(WebPLoadingContext& context, u16 color_cache_size, LittleEndianInputBitStream& bit_stream)
+{
+    // prefix-code-group     =
+    //     5prefix-code ; See "Interpretation of Meta Prefix Codes" to
+    //                  ; understand what each of these five prefix
+    //                  ; codes are for.
+
+    // "Once code lengths are read, a prefix code for each symbol type (A, R, G, B, distance) is formed using their respective alphabet sizes:
+    //  * G channel: 256 + 24 + color_cache_size
+    //  * other literals (A,R,B): 256
+    //  * distance code: 40"
+    static const size_t alphabet_sizes[] = { 256 + 24 + static_cast<size_t>(color_cache_size), 256, 256, 256, 40 };
```

Comment:
You can make this a `Array<size_t, 5>` and use `.size()` or simple iteration in the loop below